apps/nsq_to_nsq/nsq_to_nsq.go

```diff
@@ -255,32 +254,28 @@ func hasArg(s string) bool {
 	return argExist
 }
 
-func main() {
-	var selectedMode int
-
-	cCfg := nsq.NewConfig()
-	pCfg := nsq.NewConfig()
+func commandLineValidation() {
 
-	flag.Var(&nsq.ConfigFlag{cCfg}, "consumer-opt", "option to passthrough to nsq.Consumer (may be given multiple times, see http://godoc.org/github.com/nsqio/go-nsq#Config)")
-	flag.Var(&nsq.ConfigFlag{pCfg}, "producer-opt", "option to passthrough to nsq.Producer (may be given multiple times, see http://godoc.org/github.com/nsqio/go-nsq#Config)")
-
-	flag.Parse()
-
-	if *showVersion {
-		fmt.Printf("nsq_to_nsq v%s\n", version.Binary)
-		return
+	if len(topics) == 0 || *channel == "" {
+		log.Fatal("--topic and --channel are required")
 	}
 
-	if *topic == "" || *channel == "" {
-		log.Fatal("--topic and --channel are required")
+	if len(topics) == 1 {
+		if *destTopic == "" {
+			*destTopic = topics[0]
+		}
 	}
 
-	if *destTopic == "" {
-		*destTopic = *topic
+	if len(topics) > 1 {
+		if *destTopic == "" {
+			log.Fatal("--destination-topic must be specified when multi topic is enabled")
```

Comment:
... ignore my second paragraph there, I was thinking of something else :) really just ignore that.

In summary, yes, creating new topics at the destination nsqd is OK and expected.